feat: Support `VARBINARY`

hudclark opened the pull request:
# Overview
This PR adds support for encoding/decoding the `[]byte` go type as `VARBINARY` data.

# Additional Information
To keep the scope of this PR small, I did _not_ add support for reading `VARBINARY` values within complex types. This is not a regression, but a gap in the serialization/deserialization code. Timestamps, Numbers, etc also have the same issue - they are read into go types right off of the decoded JSON. This means that a `ROW(X'68656C6C6F')` is deserialized as a `[]interface{}{"aGVsbG8="}`. I did update the README to document this behavior.

To showcase this behavior, I've added a (skipped) failing test case. Given the behavior is documented, changing it would be a breaking change. Perhaps an option could be added in the future to recursively parse complex types such that this test case passes?

I'm happy to drop the commit with the skipped test case if we feel it adds noise or does not represent the desired state.

Resolves:
- https://github.com/trinodb/trino-go-client/issues/60

---

> [!NOTE]
> I have signed and emailed an individual CLA.

hudclark commented on trino/integration_test.go:
```diff
@@ -110,6 +112,14 @@ func TestMain(m *testing.M) {
 					"8080/tcp",
 					"8443/tcp",
 				},
+			}, func(hc *docker.HostConfig) {
+				hc.Ulimits = []docker.ULimit{
+					{
+						Name: "nofile",
+						Hard: 4096,
+						Soft: 4096,
+					},
+				}
```
Trino requires a minimum of 4096 descriptors in order to startup. On my linux machine, I found docker defaults to a limit of 1024.

I've updated the test configuration such that we can be sure the container is configured with reasonable ulimits.

nineinchnick commented on README.md:
```diff
@@ -317,14 +318,18 @@ structs to the `Scan()` function:
 * `map[string]interface{}` for Trino maps
-* `string` for other Trino types, as character, date, time, or timestamp
+* `string` for other Trino types, as character, date, time, or timestamp.
+  * `VARBINARY` columns are returned as base64-encoded strings
```
Is this supposed to be a one item sub-list? It looks weird, maybe add it as a note below the list.

nineinchnick reviewed (approved):
Looks great! Once the CLA is cleared, I can merge it. Please remove prefixes from commit messages.

hudclark commented:
Appreciate the quick review @nineinchnick! I've removed the commit message prefixes and updated the readme to resolve that strange formatting.

I also want to call out that this might be considered a breaking change for clients who have previously been reading in `VARBINARY` as strings, then doing base64 decoding client-side.

hudclark commented:
I will investigate and fix the failing test shortly

nineinchnick commented:
@cla-bot check

hudclark commented:
Hey @nineinchnick, looks like I've been added now https://github.com/trinodb/cla/commit/4053dbd8a21141cb5f9dbd984d0998ce5d754a7e.

Mind trying again? Thanks!

nineinchnick commented:
Released as https://github.com/trinodb/trino-go-client/releases/tag/v0.322.0
